include/nbl/builtin/hlsl/shared_memory_accessor.hlsl

```diff
@@ -0,0 +1,137 @@
+    void get(const uint ix, out float2 value) { value = asfloat(uint2(accessor.get(ix), accessor.get(ix + _NBL_HLSL_WORKGROUP_SIZE_)));}
+    void get(const uint ix, out float3 value) { value = asfloat(uint3(accessor.get(ix), accessor.get(ix + _NBL_HLSL_WORKGROUP_SIZE_), accessor.get(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_)));}
+    void get(const uint ix, out float4 value) { value = asfloat(uint4(accessor.get(ix), accessor.get(ix + _NBL_HLSL_WORKGROUP_SIZE_), accessor.get(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_), accessor.get(ix + 3 * _NBL_HLSL_WORKGROUP_SIZE_)));}
+
+    void set(const uint ix, const uint value) {accessor.set(ix, value);}
+    void set(const uint ix, const uint2 value) {
+        accessor.set(ix, value.x);
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, value.y);
+    }
+    void set(const uint ix, const uint3 value)  {
+        accessor.set(ix, value.x);
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, value.y);
+        accessor.set(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_, value.z);
+    }
+    void set(const uint ix, const uint4 value) {
+        accessor.set(ix, value.x);
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, value.y);
+        accessor.set(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_, value.z);
+        accessor.set(ix + 3 * _NBL_HLSL_WORKGROUP_SIZE_, value.w);
+    }
+
+    void set(const uint ix, const int value) {accessor.set(ix, asuint(value));}
+    void set(const uint ix, const int2 value) {
+        accessor.set(ix, asuint(value.x));
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.y));
+    }
+    void set(const uint ix, const int3 value)  {
+        accessor.set(ix, asuint(value.x));
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.y));
+        accessor.set(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.z));
+    }
+    void set(const uint ix, const int4 value) {
+        accessor.set(ix, asuint(value.x));
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.y));
+        accessor.set(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.z));
+        accessor.set(ix + 3 * _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.w));
+    }
+
+    void set(const uint ix, const float value) {accessor.set(ix, asuint(value));}
+    void set(const uint ix, const float2 value) {
+        accessor.set(ix, asuint(value.x));
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.y));
+    }
+    void set(const uint ix, const float3 value)  {
+        accessor.set(ix, asuint(value.x));
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.y));
+        accessor.set(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.z));
+    }
+    void set(const uint ix, const float4 value) {
+        accessor.set(ix, asuint(value.x));
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.y));
+        accessor.set(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.z));
+        accessor.set(ix + 3 * _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.w));
+    }
```

Comment:
what about making these templates
```cpp
template<typename T, uint32_t N>
void set(const uint32_t ix, const vector<T,N> value)
{
   [unroll(N)]
   for (uint32_t i=0; i<N; i++)
       accessor.set(ix*NumerMemoryAccessor::element_stride+i*NumerMemoryAccessor::component_stride,asuint(value[i]));
}
```

Reply:
btw if we made a
```cpp
template<typename T>
T fromuint(const uint value);
```
then we could template the `get`-ters too 

P.S. the getters need `NBL_REF_ARG` on their out value